RecoEgamma/ElectronIdentification/interface/ElectronMVAEstimatorRun2Fall17.h

```diff
@@ -0,0 +1,104 @@
+#ifndef ElectronMVAEstimatorRun2Fall17Fall17_h
+#define ElectronMVAEstimatorRun2Fall17Fall17_h
+
+#include <vector>
+#include <string>
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/EgammaCandidates/interface/Conversion.h"
+#include "DataFormats/EgammaCandidates/interface/ConversionFwd.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "RecoEgamma/EgammaTools/interface/AnyMVAEstimatorRun2Base.h"
+#include "RecoEgamma/EgammaTools/interface/ConversionTools.h"
+#include "RecoEgamma/EgammaTools/interface/GBRForestTools.h"
+
+class ElectronMVAEstimatorRun2Fall17 : public AnyMVAEstimatorRun2Base{
+
+ public:
+
+  // Constructor and destructor
+  ElectronMVAEstimatorRun2Fall17(const edm::ParameterSet& conf, bool withIso);
+  ~ElectronMVAEstimatorRun2Fall17() override;
+
+  // Calculation of the MVA value (VID accessor)
+  float mvaValue( const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const override;
+  // Calculation of the MVA value (fwlite-compatible accessor)
+  float mvaValue( const reco::GsfElectron * particle, const edm::EventBase &) const ;
+  // Calculation of the MVA value (bare version)
+  float mvaValue( const int iCategory, const std::vector<float> & vars) const ;
+
+  // Utility functions
+  int getNCategories() const override { return nCategories; }
+  bool isEndcapCategory( int category ) const;
+  //const std::string& getName() const final { return name_; }
+  const std::string& getTag() const final { return tag_; }
+
+  // Functions that should work on both pat and reco electrons
+  // (use the fact that pat::Electron inherits from reco::GsfElectron)
+  std::vector<float> fillMVAVariables(const edm::Ptr<reco::Candidate>& particle, const edm::Event&) const override;
+  std::vector<float> fillMVAVariables( const reco::GsfElectron * particle, const edm::Handle<reco::ConversionCollection> conversions, const reco::BeamSpot *beamSpot, const edm::Handle<double> rho) const;
+  int findCategory( const edm::Ptr<reco::Candidate>& particle) const override;
+  int findCategory( const reco::GsfElectron * particle) const ;
+  // The function below ensures that the variables passed to MVA are
+  // within reasonable bounds
+  void constrainMVAVariables(std::vector<float>&) const;
+
+  // Call this function once after the constructor to declare
+  // the needed event content pieces to the framework
+  void setConsumes(edm::ConsumesCollector&&) const override;
+  // Call this function once per event to retrieve all needed
+  // event content pices
+
+ protected:
+
+  // Define here the number and the meaning of the categories
+  // for this specific MVA
+  const int nCategories = 6;
+  const int NVar = 22;
```

Comment:
private (+ protected) member variables end with an underscore. Additionally the first letter should be lowercase